cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -375,9 +382,63 @@ func (o *Options) Run(ctx context.Context) error {
 		return err
 	}
 
+	if err := server.GenericAPIServer.AddPostStartHook("bootstrap-public-info", func(_ genericapiserver.PostStartHookContext) error {
+		if err := createNamespaces(gardencorev1beta1.GardenerSystemPublicNamespace); err != nil {
```

Comment:
As per https://github.com/kubernetes/kubernetes/blob/2bda5dd8c794d933145938affca30f012986e8fa/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L183-L184

> 	// PostStartHooks are each called after the server has started listening, in a separate go func for each
>	// with no guarantee of ordering between them.

Thus, we must not rely on the `bootstrap-garden-cluster` to run before `bootstrap-public-info` and ensure the namespace `gardener-system-public` is created before we try to create the configmap in this namespace.